aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -2060,6 +2060,20 @@ bool cpu_equal(const Tensor& self, const Tensor& other) {
   if (!self.is_same_size(other)) {
     return false;
   }
+  // Since the flags like neg/conj should be already handled outside the
+  // TensorIterator, it should be safe to have the following fast path by
+  // ensuring the storage and strides exactly the same.
+  if (self.dtype() == other.dtype()
+      && self.sizes().equals(other.sizes())
+      && self.strides().equals(other.strides())
```

Comment:
A fastpath for this would be to instead assert both tensors are contiguous, before checking their strides